tests/tests_geomstats/test_spd_matrices.py

```diff
@@ -19,559 +17,676 @@
+                        [1 / 3, 0.125, 0.125],
+                        [1 / 3, 0.125, 0.125],
+                    ],
+                    base_point=[[1.0, 0.0, 0.0], [0.0, 2.5, 1.5], [0.0, 1.5, 2.5]],
+                    expected=[[2.0, 1.0, 1.0], [1.0, 0.5, 0.5], [1.0, 0.5, 0.5]],
+                )
+            ]
+            return self.generate_tests(smoke_data)
+
+        def differential_log_data(self):
+            smoke_data = [
+                dict(
+                    tangent_vec=[[1.0, 1.0, 3.0], [1.0, 1.0, 3.0], [3.0, 3.0, 4.0]],
+                    base_point=[[1.0, 0.0, 0.0], [0.0, 1.0, 0.0], [0.0, 0.0, 4.0]],
+                    expected=[
+                        [1.0, 1.0, 2 * LN_2],
+                        [1.0, 1.0, 2 * LN_2],
+                        [2 * LN_2, 2 * LN_2, 1],
+                    ],
+                )
+            ]
+            return self.generate_tests(smoke_data)
+
+        def inverse_differential_log_data(self):
+            smoke_data = [
+                dict(
+                    tangent_vec=[
+                        [1.0, 1.0, 2 * LN_2],
+                        [1.0, 1.0, 2 * LN_2],
+                        [2 * LN_2, 2 * LN_2, 1],
+                    ],
+                    base_point=[[1.0, 0.0, 0.0], [0.0, 1.0, 0.0], [0.0, 0.0, 4.0]],
+                    expected=[[1.0, 1.0, 3.0], [1.0, 1.0, 3.0], [3.0, 3.0, 4.0]],
+                )
+            ]
+
+            return self.generate_tests(smoke_data)
+
+        def differential_exp_data(self):
+            smoke_data = [
+                dict(
+                    tangent_vec=[[1.0, 1.0, 1.0], [1.0, 1.0, 1.0], [1.0, 1.0, 1.0]],
+                    base_point=[[1.0, 0.0, 0.0], [0.0, 1.0, 0.0], [0.0, 0.0, -1.0]],
+                    expected=[
+                        [EXP_1, EXP_1, SINH_1],
+                        [EXP_1, EXP_1, SINH_1],
+                        [SINH_1, SINH_1, 1 / EXP_1],
+                    ],
+                )
+            ]
+            return self.generate_tests(smoke_data)
+
+        def inverse_differential_exp_data(self):
+            smoke_data = [
+                dict(
+                    tangent_vec=[
+                        [EXP_1, EXP_1, SINH_1],
+                        [EXP_1, EXP_1, SINH_1],
+                        [SINH_1, SINH_1, 1 / EXP_1],
+                    ],
+                    base_point=[[1.0, 0.0, 0.0], [0.0, 1.0, 0.0], [0.0, 0.0, -1.0]],
+                    expected=[[1.0, 1.0, 1.0], [1.0, 1.0, 1.0], [1.0, 1.0, 1.0]],
+                )
+            ]
+            return self.generate_tests(smoke_data)
+
+    testing_data = TestDataSPDMatrices()
+
+    def test_belongs(self, n, mat, expected):
+        self.assertAllClose(SPDMatrices(n).belongs(gs.array(mat)), gs.array(expected))
+
+    def test_random_point(self, n, num_points):
+        space = SPDMatrices(n)
+        self.assertAllClose(gs.all(space.random_point(num_points)), gs.array(True))
```

Comment:
What are you testing here? Should we rather test the shape of the random points ? or random_point_and_belongs?